src/INode.java

```diff
@@ -244,14 +320,38 @@ private ASTValue evalExponentiate(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");	
 	}
 
-	private ASTValue evalGEQ(WordsEnvironment environment) {
-		// TODO
-		throw new AssertionError("Not yet implemented");	
+	private ASTValue evalGEQ(WordsEnvironment environment) throws WordsRuntimeException {
+		ASTValue lhs = children.get(0).eval(environment);
+		ASTValue rhs = children.get(1).eval(environment);
+		
+		checkRelOpArgTypes(lhs, rhs);
+		
+		// lhs and rhs are now the same type
+		switch (lhs.type) {
+			case NUM:
+				return new ASTValue(lhs.numValue >= rhs.numValue);
```

Comment:
Once again, some error checking for the equals part of geq. (Same comment applies to leq).